core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -193,8 +215,18 @@ private[spark] object HighlyCompressedMapStatus {
     } else {
       0
     }
+    val hugeBlockSizesArray = ArrayBuffer[Tuple2[Int, Byte]]()
+    if (numNonEmptyBlocks > 0) {
+      uncompressedSizes.zipWithIndex.foreach {
+        case (size, reduceId) =>
+          if (size > 2 * avgSize) {
```

Comment:
This is yet to be addressed - the scale constant `2`, and putting a lower bound on `size`